packages/server/src/oauth/authorize.ts

```diff
@@ -128,6 +114,7 @@ async function validateAuthorizeRequest(req: Request, res: Response, params: Rec
       codeChallenge: params.code_challenge ?? existingLogin.codeChallenge,
       codeChallengeMethod: params.code_challenge_method ?? existingLogin.codeChallengeMethod,
       code: generateSecret(16),
+      launch: params.launch ? { reference: `SmartAppLaunch/${params.launch}` } : existingLogin.launch,
```

Comment:
Unfortunately I think we need to validate that `params.launch` is actually a valid ID of a `SmartAppLaunch` resource